src/main/kotlin/com/avail/descriptor/fiber/FiberDescriptor.kt

```diff
@@ -1001,44 +980,26 @@ class FiberDescriptor private constructor(
 	override fun o_DebugLog(self: AvailObject): StringBuilder =
 		self.mutableSlot(DEBUG_LOG).javaObjectNotNull()
 
-	override fun o_Lock(self: AvailObject, critical: Continuation0) {
+	override fun <T> o_Lock(self: AvailObject, supplier: ()->T): T =
 		when (val interpreter = Interpreter.currentOrNull()) {
 			null -> {
 				// It's not running an AvailThread, so don't bother detecting
 				// multiple nested fiber locks (which would suggest a deadlock
 				// hazard)..
-				synchronized(self) { critical.value() }
+				synchronized(self) { supplier() }
 			}
 			else -> {
-				interpreter.lockFiberWhile(
-					self,
-					Supplier {
-						synchronized(self) { critical.value() }
-					})
-			}
-		}
-	}
-
-	override fun <T> o_Lock(self: AvailObject, supplier: Supplier<T>): T =
-		when (val interpreter = Interpreter.currentOrNull()) {
-			null -> {
-				// It's not running an AvailThread, so don't bother detecting
-				// multiple nested fiber locks (which would suggest a deadlock
-				// hazard)..
-				synchronized(self) { supplier.get() }
-			}
-			else -> {
-				interpreter.lockFiberWhile(
-					self,
-					Supplier<T> { synchronized(self) { supplier.get() } })
+				interpreter.lockFiberWhile(self) {
+					synchronized(self) { supplier() }
+				}
 			}
 		}
 
-	override fun mutable(): FiberDescriptor = mutable
+	override fun mutable(): AbstractDescriptor = mutable
```

Comment:
We can leave off the type declaration.  The val fields are local to the companion, so Kotlin won't complain about "platform operation" result types.

Reply:
Whoops, yep, thought I got most of these. I'll do a regex right now to fix this systematically, so don't waste your time documenting more of them.

Reply:
Okay, nailed them. And everything still compiles and passes the tests, so the regex was good. 👍